### PR TITLE
Add preliminary Conway support

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

File: cardano-lib/mainnet-config.nix [hunk 1]
```diff
@@ -12,7 +12,8 @@
   ShelleyGenesisHash = "1a3be38bcbb7911969283716ad7aa550250226b76a61fc51cc9a9a35d9276d81";
   AlonzoGenesisFile = ./mainnet + "/alonzo-genesis.json";
   AlonzoGenesisHash = "7e94a15f55d1e82d10f09203fa1d40f8eede58fd8066542cf6566008068ed874";
-
+  ConwayGenesisFile = ./mainnet + "/alonzo-genesis.json";
```
Comment (disassembler) [inline cardano-lib/mainnet-config.nix:15]: Should be conway

File: cardano-lib/marlowe-pioneers-config.nix [hunk 2]
```diff
@@ -12,6 +12,8 @@
   ShelleyGenesisHash = "10658475f00452b93d4f1ac083f2b55b12585a20d2fb505d60c8a03dd6f46ae5";
   AlonzoGenesisFile = ./marlowe-pioneers + "/alonzo-genesis.json";
   AlonzoGenesisHash = "7e94a15f55d1e82d10f09203fa1d40f8eede58fd8066542cf6566008068ed874";
+  ConwayGenesisFile = ./mainnet + "/alonzo-genesis.json";
```
Comment (disassembler) [inline cardano-lib/marlowe-pioneers-config.nix:15]: Should be conway

File: cardano-lib/p2p-config.nix [hunk 3]
```diff
@@ -12,6 +12,8 @@
   ShelleyGenesisHash = "a0e8e5520ab7c452c4e36020ded12c791a10abd7ca25c083af6149fe269ddb67";
   AlonzoGenesisFile = ./p2p + "/alonzo-genesis.json";
   AlonzoGenesisHash = "7e94a15f55d1e82d10f09203fa1d40f8eede58fd8066542cf6566008068ed874";
+  ConwayGenesisFile = ./mainnet + "/alonzo-genesis.json";
```
Comment (disassembler) [inline cardano-lib/p2p-config.nix:15]: Should be conway

File: cardano-lib/preprod-config.nix [hunk 4]
```diff
@@ -11,6 +11,8 @@
   ShelleyGenesisHash = "162d29c4e1cf6b8a84f2d692e67a3ac6bc7851bc3e6e4afe64d15778bed8bd86";
   AlonzoGenesisFile = ./preprod + "/alonzo-genesis.json";
   AlonzoGenesisHash = "7e94a15f55d1e82d10f09203fa1d40f8eede58fd8066542cf6566008068ed874";
+  ConwayGenesisFile = ./mainnet + "/alonzo-genesis.json";
```
Comment (disassembler) [inline cardano-lib/preprod-config.nix:14]: Should be conway

File: cardano-lib/preview-config.nix [hunk 5]
```diff
@@ -10,6 +10,8 @@
   ShelleyGenesisHash = "363498d1024f84bb39d3fa9593ce391483cb40d479b87233f868d6e57c3a400d";
   AlonzoGenesisFile = ./preview + "/alonzo-genesis.json";
   AlonzoGenesisHash = "7e94a15f55d1e82d10f09203fa1d40f8eede58fd8066542cf6566008068ed874";
+  ConwayGenesisFile = ./mainnet + "/alonzo-genesis.json";
```
Comment (disassembler) [inline cardano-lib/preview-config.nix:13]: Should be conway

File: cardano-lib/shelley_qa-config.nix [hunk 6]
```diff
@@ -12,6 +12,8 @@
   ShelleyGenesisHash = "85d1783750753deaa6560158eb85fcb30078ea32a36e12dd8af39168bc053d09";
   AlonzoGenesisFile = ./shelley_qa + "/alonzo-genesis.json";
   AlonzoGenesisHash = "7e94a15f55d1e82d10f09203fa1d40f8eede58fd8066542cf6566008068ed874";
+  ConwayGenesisFile = ./mainnet + "/alonzo-genesis.json";
```
Comment (disassembler) [inline cardano-lib/shelley_qa-config.nix:15]: Should be conway

File: cardano-lib/sre.nix [hunk 7]
```diff
@@ -12,6 +12,8 @@
   ShelleyGenesisHash = "bdf17f12d82d89908211f693c30e1cb114fe4b0954159394f08ca043c3446b95";
   AlonzoGenesisFile = ./sre + "/alonzo-genesis.json";
   AlonzoGenesisHash = "fd77bbad445e6c438e2755d5e939a818d5d231316f882c7725988ebfac8442f8";
+  ConwayGenesisFile = ./mainnet + "/alonzo-genesis.json";
```
Comment (disassembler) [inline cardano-lib/sre.nix:15]: Should be conway

File: cardano-lib/staging-config.nix [hunk 8]
```diff
@@ -12,7 +12,8 @@
   ShelleyGenesisHash = "cbb0f57a001120ef11d97b943ce6953b75d26adc11dc68451e9dec7989a62c85";
   AlonzoGenesisFile = ./staging + "/alonzo-genesis.json";
   AlonzoGenesisHash = "7e94a15f55d1e82d10f09203fa1d40f8eede58fd8066542cf6566008068ed874";
-
+  ConwayGenesisFile = ./mainnet + "/alonzo-genesis.json";
```
Comment (disassembler) [inline cardano-lib/staging-config.nix:15]: Should be conway

File: cardano-lib/testnet-config.nix [hunk 9]
```diff
@@ -12,7 +12,8 @@
   ShelleyGenesisHash = "849a1764f152e1b09c89c0dfdbcbdd38d711d1fec2db5dfa0f87cf2737a0eaf4";
   AlonzoGenesisFile = ./testnet + "/alonzo-genesis.json";
   AlonzoGenesisHash = "7e94a15f55d1e82d10f09203fa1d40f8eede58fd8066542cf6566008068ed874";
-
+  ConwayGenesisFile = ./mainnet + "/alonzo-genesis.json";
```
Comment (disassembler) [inline cardano-lib/testnet-config.nix:15]: Should be conway

File: cardano-lib/vasil-dev-config.nix [hunk 10]
```diff
@@ -11,6 +11,8 @@
   ShelleyGenesisHash = "25264c807eb610ca6c0706fa051a4cefbf37c9b9bbea9f669a0b085f9e3e840a";
   AlonzoGenesisFile = ./vasil-dev + "/alonzo-genesis.json";
   AlonzoGenesisHash = "8bfa9249601f7174594b0e792a4c5697421df1ec8102a236224b4e0e398dd1d7";
+  ConwayGenesisFile = ./mainnet + "/alonzo-genesis.json";
```
Comment (disassembler) [inline cardano-lib/vasil-dev-config.nix:14]: Should be conway

**Review by disassembler — CHANGES_REQUESTED** [review]
This is missing conway genesis JSON file added for each environment. It also references mainnet directory for all environments.

**Comment by erikd** [comment]
I have fixed what you pointed out as obviously wrong. I do not know how to fix the rest (or test that its right).

**Review by angerman — APPROVED** [review]

**Comment by angerman** [comment]
Added "(dummy)" to the Conway genesis link, just to make sure the casual reader is not confused.

**Comment by angerman** [comment]
bors try

**Pull request merged**